apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/RowFilters.cs

```diff
@@ -0,0 +1,380 @@
+        /// The range of columns from which cells should be matched. Must not be null.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter ColumnRange(ColumnRange range) =>
+            new RowFilter { ColumnRangeFilter = GaxPreconditions.CheckNotNull(range, nameof(range)) };
+
+        // TODO: Add additional ColumnRange... methods to take start/end values?
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which applies one of two
+        /// possible RowFilters to the data based on the output of a predicate
+        /// RowFilter.
+        /// </summary>
+        /// <param name="predicateFilter">
+        /// If <paramref name="predicateFilter"/> outputs any cells, then
+        /// <paramref name="trueFilter"/> will be evaluated on the input row.
+        /// Otherwise, <paramref name="falseFilter"/> will be evaluated.
+        /// Must not be null.
+        /// </param>
+        /// <param name="trueFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// returns any results. If null, no results will be returned in the true case.
+        /// </param>
+        /// <param name="falseFilter">
+        /// The filter to apply to the input row if <paramref name="predicateFilter"/>
+        /// does not return any results. If null, no results will be returned in the
+        /// false case.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter Condition(
+            RowFilter predicateFilter, RowFilter trueFilter, RowFilter falseFilter) =>
+            new RowFilter
+            {
+                Condition = new RowFilter.Types.Condition
+                {
+                    // TODO: Verify this on server. Might be allowed for (row contains any cells).
+                    PredicateFilter = GaxPreconditions.CheckNotNull(predicateFilter, nameof(predicateFilter)),
+                    TrueFilter = trueFilter,
+                    FalseFilter = falseFilter
+                }
+            };
+
+        /// <summary>
+        /// Creates a new <see cref="RowFilter"/> instance which matches only cells
+        /// from columns whose families satisfy the given RE2 regex.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// For technical reasons, the regex must not contain the `:` character, even
+        /// if it is not being used as a literal. Note that, since column families
+        /// cannot contain the new line character `\n`, it is sufficient to use `.`
+        /// as a full wildcard when matching column family names.
+        /// </para>
+        /// </remarks>
+        /// <param name="regex">
+        /// The RE2 regex used to match column families. Must not be null.
+        /// </param>
+        /// <returns>The created row filter.</returns>
+        public static RowFilter FamilyNameRegex(string regex) =>
```

Comment:
It's odd that this accepts a string but `ColumnQualifierRegex` accepts a `BigtableByteString`. Presumably that's just the way the API is specified, but we should *consider* removing the inconsistency at this level. I'm slightly torn.

Reply:
I'm asking about this internally.

Reply:
And yes, that's for the API is specified by the way.